.github/workflows/vulnerability-scanner-tests.yml

```diff
@@ -0,0 +1,96 @@
+name: Vulnerability Scanner
+
+on:
+  pull_request:
+    # Pull request events
+    types: [synchronize, opened, reopened, ready_for_review]
+    
+    # Path filtering
+    paths:
+      - ".github/workflows/vulnerability-scanner-tests.yml"
+      - "src/shared_modules/router/**"
+      - "src/shared_modules/content_manager/**"
+      - "src/shared_modules/indexer_connector/**"
+      - "src/wazuh_modules/vulnerability_scanner/**"
+      - "src/wazuh_modules/wazuh-http-request/**"
+
+jobs:
+  vulnerability-scanner-modules:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 2 # We need to fetch 2 commits to get the diff.
```

Comment:
What would happen if I push more than two commits?
```
...
git commit -m "Commit a"
...
git commit -m "Commit b"
...
git commit -m "Commit c"
git push
```

Reply:
Good question.. maybe I should check the diffs against the base head?

Reply:
Not quite sure. If the trigger is a pull request, things get complicated. 
Maybe using [GitHub CLI](https://cli.github.com/manual/gh_pr)? [Here](https://github.com/godotengine/godot/blob/master/.github/workflows/static_checks.yml#L34) is an example of how that could be achieved.

Reply:
Found a generic solution:
`git diff --name-only origin/${{ github.head_ref } origin/${{ github.base_ref }})`

github.head_ref -> The HEAD of the current branch
github.base_ref -> The HEAD of the branch where the PR points

Reply:
Seems like I was wrong. Also, to apply your suggestion we need to have a token from the repo admins :/